server_development/topics/providers.adoc

```diff
@@ -527,6 +527,78 @@ ifeval::["{kc_dist}" == "wildfly"]
 Once you have a JAR file with a descriptor and the scripts you want to deploy, you just need to copy the JAR to the {project_name} `standalone/deployments/` directory.
 endif::[]
 
+===== Deploy the script engine on Java 15 and later
+
+In order to run the scripts, JavaScript providers require JavaScript engine to be available in your Java application. In the Java 14 or earlier, there is Nashorn JavaScript Engine
+automatically available as part of the Java itself and JavaScript providers are able to leverage this script engine out of the box. However in case of Java 15 or later, script engine is not part
+of the Java itself and needs to be added to your server. {project_name} does not have any script engine by default. So in case of Java 15 or later, there is additional step needed when deploying script
+providers - adding the script engine of your choice to your distribution.
+
+You can use any script engine you want, however we test just with the Nashorn JavaScript Engine and in the steps below, it is assumed that this one will be used:
+
+ifeval::["{kc_dist}" == "quarkus"]
+You can install script engine by copying the nashorn script engine JAR together with it's dependencies directly to the `KEYCLOAK_HOME/providers` directory. In the `pom.xml` file
+of your script project, you can declare the dependency like this in the `dependencies` section:
+
+```xml
+<dependency>
+    <groupId>org.openjdk.nashorn</groupId>
+    <artifactId>nashorn-core</artifactId>
+    <version>15.3</version>
+</dependency>
+```
+
+and declare `maven-dependency-plugin` in the `plugins` section to copy the dependencies to the specified directory:
+
+```xml
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-dependency-plugin</artifactId>
+    <executions>
+        <execution>
+                <id>copy-dependencies-quarkus</id>
+                <phase>package</phase>
+            <goals>
+                <goal>copy-dependencies</goal>
+            </goals>
+            <configuration>
+                <outputDirectory>${project.build.directory}/keycloak-server-copy/providers</outputDirectory>
+                <includeArtifactIds>nashorn-core,asm,asm-util,asm-commons</includeArtifactIds>
+            </configuration>
+        </execution>
+    </executions>
+</plugin>
+```
+Once the project is built, you can copy the script engine together with it's dependencies to the `KEYCLOAK_HOME/providers` directory.
+```bash
+cp target/keycloak-server-copy/providers/*.jar KEYCLOAK_HOME/providers/
+```
+After re-augment the distribution with `kc.sh build` the script engine should be deployed and your script providers should work.
```

Comment:
```suggestion
After re-augment, the distribution with `kc.sh build` the script engine should be deployed and your script providers should work.
```

Reply:
I am not sure about this one change. Let me explain: re-augmenting of the distribution itself is done by running `kc.sh build` script. So it is same like if I say for example:
`After changing the house with new bricks` (Probably dummy example, but I don't have any better in my mind :-) ).

So I wonder if the comma should be before the second `the` ? For example:
`After re-augment the distribution with `kc.sh build`, the script engine should be deployed and your script providers should work.`

WDYT?

Reply:
Exactly.  I was just looking at this and coming to the same conclusion.  Thanks for correcting this, @mposolda.